modules/actor/actor-wfrp4e.js

```diff
@@ -1368,7 +1368,7 @@ export default class ActorWfrp4e extends Actor {
       cardOptions.speaker.scene = canvas.scene.id
       cardOptions.flags.img = this.token.texture.src; // Use the token image instead of the actor image
 
-      if (this.token.hidden) {
+      if (this.token.document?.hidden ?? this.token.hidden) {
         cardOptions.speaker.alias = "???"
         cardOptions.flags.img = "systems/wfrp4e/tokens/unknown.png"
       }
```

Comment:
I removed this in https://github.com/moo-man/WFRP4e-FoundryVTT/commit/7cd770ad8f435ed2d8e18cc7a97ff5e721fb5b57

`this.token` refers to a `TokenDocument` instance already, what's the purpose of checking for `document`. It's always undefined. 
![image](https://user-images.githubusercontent.com/28637157/221316007-dc0831f0-a319-47a1-9159-7f7f5ac94d37.png)

Reply:
I'm pretty sure, that i was getting some invalid data here - that this.token.hidden sometimes was not present, but maybe that's because im not so fluent with Foundry Api. Another option it is related to Token Action hud behavior. I will check that and if reproducable - add another comment or remove the change.

Reply:
Any update on this? Would like to merge soon

Reply:
i removed that, couldnt reproduce anymore on clean system, must have been some other changes or modules i had.